homeassistant/components/rainmachine/coordinator.py

```diff
@@ -0,0 +1,99 @@
+"""Define RainMachine utilities."""
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from datetime import timedelta
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect,
+    async_dispatcher_send,
+)
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import LOGGER
+
+SIGNAL_REBOOT_COMPLETED = "rainmachine_reboot_completed_{0}"
+SIGNAL_REBOOT_REQUESTED = "rainmachine_reboot_requested_{0}"
+
+
+class RainMachineDataUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
```

Comment:
The type change from `dict` to `dict[str, Any]` seems to be the cause for mypy failing. Can you please revert this and check via `pre-commit` if mypy then doesn't issue errors? And it would be good to fix the typing stuff in a future PR.